src/mol-model-formats/structure/pdb/atom-site.ts

```diff
@@ -91,7 +91,7 @@ export function getAtomSite(sites: AtomSiteTemplate, hasTer: boolean): { [K in k
         if (asymIdCounts.has(asymId)) {
             // only change the chains name if there are TER records
             // otherwise assume repeated chain name use is from interleaved chains
-            if (hasTer && asymIdChanged) {
+            if (terIndices.size > 0 && asymIdChanged || terIndices.has(i)) {
```

Comment:
```suggestion
            if ((terIndices.size > 0 && asymIdChanged) || terIndices.has(i)) {
```

Alternatively, would it now be enough to just check for ``terIndides.has(i)`` ?

Reply:
yeah, should be good